src/classes/CW_CampaignTriggerHandler.cls

```diff
@@ -0,0 +1,308 @@
+/**
+* File Name   :    CW_CampaignTriggerHandler.cls
+* Description :    B-108780: Lock the creation of new campaign for a roadmap that 
+*				   is "In Approval" or "Approved"
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 16/04/2014 Cristina P.	Intial
+* 0.2 29/04/2014 Chris Yorskton MarCom Points on Campaign rolled up to Roadmap
+* 0.3 21/05/2014 Chris Yorskton Trigger on Campaign, every time stage changes, update stage (picklist) on all child schedules
+*
+*/
+
+//Cristina P: 25/06/14: Changing the class from "with sharing" to "without sharing" 
+//so the user can create related records while cost/brief is in approval
+public without sharing class CW_CampaignTriggerHandler 
+{
+	
+	private Boolean m_isExecuting = false;
+
+    public CW_CampaignTriggerHandler(Boolean isExecuting){
+        m_isExecuting = isExecuting;
+    }
+	
+	public void OnBeforeInsert(
+		List<CW_Campaign__c> newObjects, 
+		Map<Id,CW_Campaign__c> newObjectMap
+	)
+	{
+		stopNewCampaignCreationWhenRoadmapLocked(newObjects);
+	}
+	
+	public void OnAfterInsert( CW_Campaign__c[] newObjects, Map<Id,CW_Campaign__c> newObjectMap ) {
+		rollupPointsOnCampaignToRoadmap(newObjects);
+		createSingleSchedules(newObjects);
+    }
+    
+	public void OnBeforeUpdate( Map<Id,CW_Campaign__c> oldObjectMap, Map<Id,CW_Campaign__c> newObjectMap )
```

Comment:
Methods should use the `lowerCamelCase` naming convention. I suggest to rename accordingly.